x-pack/plugins/ml/public/application/explorer/anomaly_context_menu.tsx

```diff
@@ -103,6 +136,183 @@ export const AnomalyContextMenu: FC<AnomalyContextMenuProps> = ({
 
+  const isMaxSeriesToPlotValid =
+    maxSeriesToPlot >= 1 && maxSeriesToPlot <= MAX_ANOMALY_CHARTS_ALLOWED;
+
+  const jobIds = selectedJobs.map(({ id }) => id);
+
+  const getEmbeddableInput = useCallback(
+    (timeRange?: TimeRange) => {
+      // Respect the query and the influencers selected
+      // If no query or filter set, filter out to the lanes the selected cells
+      // And if no selected cells, show everything
+
+      const selectionInfluencers = getSelectionInfluencers(
+        selectedCells,
+        selectedCells?.viewByFieldName!
+      );
+
+      const influencers = selectionInfluencers ?? [];
+      const config = getDefaultEmbeddablePanelConfig(jobIds, queryString);
+      const queryFromSelectedCells = influencers
+        .map((s) => escapeKueryForFieldValuePair(s.fieldName, s.fieldValue))
+        .join(' or ');
+
+      // When adding anomaly charts to Dashboard, we want to respect the Dashboard's time range
+      // so we are not passing the time range here
+      return {
+        ...config,
+        ...(timeRange ? { timeRange } : {}),
+        jobIds,
+        maxSeriesToPlot: maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT,
+        severityThreshold: severity.val,
+        ...((isDefined(queryString) && queryString !== '') ||
+        (queryFromSelectedCells !== undefined && queryFromSelectedCells !== '')
+          ? {
+              query: {
+                query: queryString === '' ? queryFromSelectedCells : queryString,
+                language: SEARCH_QUERY_LANGUAGE.KUERY,
+              } as Query,
+            }
+          : {}),
+      };
+    },
+    [jobIds, maxSeriesToPlot, severity, queryString, selectedCells]
+  );
+
+  const onSaveCallback: SaveModalDashboardProps['onSave'] = useCallback(
+    ({ dashboardId }) => {
+      const stateTransfer = embeddable!.getStateTransfer();
+
+      const embeddableInput: Partial<AnomalyChartsEmbeddableInput> = getEmbeddableInput();
+
+      const state = {
+        input: embeddableInput,
+        type: ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+      };
+
+      const path = dashboardId === 'new' ? '#/create' : `#/view/${dashboardId}`;
+
+      stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
+        state,
+        path,
+      });
+    },
+    [embeddable, getEmbeddableInput]
+  );
+
+  const panels = useMemo<EuiContextMenuProps['panels']>(() => {
+    const rootItems: EuiContextMenuPanelItemDescriptor[] = [];
+    const menuPanels: EuiContextMenuPanelDescriptor[] = [{ id: 'panelActions', items: rootItems }];
+
+    const getContent = (callback: ReactEventHandler<HTMLButtonElement>) => (
+      <EuiPanel paddingSize={'s'}>
+        <EuiSpacer size={'s'} />
+        <EuiForm>
+          <EuiFormRow
+            isInvalid={!isMaxSeriesToPlotValid}
+            error={
+              !isMaxSeriesToPlotValid ? (
+                <FormattedMessage
+                  id="xpack.ml.anomalyChartsEmbeddable.maxSeriesToPlotError"
```

Comment:
Nit - this file contains lots of `FormattedMessage` with different styles of prefix. Should they be consistent within the same file?